platform/engineering/collab-cycle/architecture-intent-template.md

```diff
@@ -0,0 +1,79 @@
+
+## When to schedule an Architecture Intent meeting?
+
+You should schedule an Architecture Intent meeting if any of these apply:
+
+- You have questions about one or more points on the template below.
+- You're having trouble locating or getting technical info from other stakeholders or system owners.
+- You're using an architecture pattern not currently found on the va.gov platform.
+- You plan to use a new technology, library or dependency.
+- You're integrating with a new system or API, inside or outside of the VA.
+- You need to coordinate your change with other teams or your change will add work or responsiblity to other teams.
+- You're gathering new [PII]/[PHI] or saving PII/PHI in a new place.
+- There are cost, performance or security implications to your change.
+
+[PII]: https://en.wikipedia.org/wiki/Personal_data
+[PHI]: https://en.wikipedia.org/wiki/Protected_health_information
+
+## Preparing for the Architecture Intent meeting
+
+You should write up your Architecture Intent using the template below.  The Governance Team will need at least **2 business days** to review your materials.
+
+## Architecture Intent Template
+
+Your document should be brief and high-level.  One to two pages is appropriate for many changes.  Focus on the high level and link to supporting material where appropriate; this is _not_ a detailed engineering spec.
+
+Some of the items below may not apply to your work--that's okay.  You may not be able to fill in some items that _do_ apply to your work--that's also okay.  If you don't have answers, please come ready to ask questions.
+
+- Product description
+    + Brief overview of motivation for the change
+    + Link to product doc or GitHub issue
+- UX design description
+    + For user-facing changes
+    + Link to lo-fi prototype or wireframes
+- Frontend changes
+    + Identify any significant code changes
+    + Describe any product analytics being gathered
+- Internal API changes
+    + List new or modified APIs in `vets-api`
+    + Describe expected call patterns
+- External API changes
+    + List new or modified APIs for upstream systems
+    + Describe expected call patterns
+- Background jobs
+    + List any required background processing
+    + Describe error and dead letter handling
+- Data storage
+    + Describe new or modified databases, tables or columns
+    + Describe indexes and constraints
+    + Identify PII and PHI
+- Libraries and dependencies
+    + List new or updated dependences
+- Metrics, logging, observability, alerting
+    + Identify key areas to monitor
+- Infrastructure and network changes
+    + List any changes or additions
+- Test plan
+    + Describe automated testing
+    + Describe required test data and test user accounts
+    + Note any manual tests or user acceptance tests
+- Rollout plan
```

Comment:
A main complaint I've heard that we have documentation scattered everywhere, so it might be nice think about the location of documentation at this point, so we can start to redirect teams to a more central location

Reply:
@timwright12 Strongly agree about the docs scattered everywhere.  Kicking that can down the road for this first draft, but thinking that markdown docs in `va.gov-teams` is probably the most practical central location.  I'll be soliciting opinions on this.

@sanjabaj2 Agree this is a lot for one page.  I want to find a balance between keeping this brief and making sure teams have thought through all the important things.  Changed "Test plan" to "Test strategy" and condensed to two bullet points.